Source/ThirdParty/ik/CMakeLists.txt

```diff
@@ -38,19 +34,20 @@ option (IK_MEMORY_DEBUGGING "Global switch for memory options. Keep track of the
 cmake_dependent_option (IK_MEMORY_BACKTRACE "Generate backtraces for every malloc(), making it easy to track down memory leaks" "${DEFAULT_MEMORY_DEBUGGING}" "IK_MEMORY_DEBUGGING AND NOT WEB" FALSE)
 
 # Enable restrict keyword in quaternion and vector operations if not in debug
-set (IK_RESTRICT "")
-if (NOT CMAKE_BUILD_TYPE STREQUAL Debug)
+if (NOT IK_RESTRICT)
+    set (IK_RESTRICT "")
```

Comment:
If IK_RESTRICT has an empty value it will always do the check, regardless of whether it's been cached or not (I don't know of any compilers that don't support the restrict keyword however, but this would still be useful to know for other CMake stuff).

This prints "lol" every time you run it:
```cmake
cmake_minimum_required (VERSION 3.3)
project (lalala)
if (NOT lol)
    set (lol ${lol} CACHE STRING "")
    message (STATUS "lol")
endif ()
```